### PR TITLE
[CR] New Town Hall Variant

### PR DESCRIPTION
**Pull request by JonathanLochridge**
#### Summary
Content "New Town Hall Variant"

#### Purpose of change
I thought having more town hall variants would be nice. Also that it might be a relatively easy thing to change for a first location change.

#### Describe the solution
Changed the classroom on floor 2 to a set of conference rooms.
Made the windows more symmetrical.

#### Describe alternatives you've considered
-A variant with an included small library.
-A variant with an attached greenspace and parking lot.

#### Testing
All files Linted.
New variant can spawn in latest experimental when these files are added.
Error occurs on startup but I think it is unrelated to the new variant.

#### Additional context
It could have some minor items added to the new room. Trying to figure out how to do that. Now that I got it to spawn properly I should work on that and adding it to the start location for the town hall on my next pass after this PR.
Functional

### CODE REVIEW

File: data/json/mapgen/town_hall.json [hunk 1]
```diff
@@ -75,6 +75,126 @@
...
+      ],
+      "palettes": [ "town_hall_palette" ],
+      "place_monsters": [
+        { "monster": "GROUP_ZOMBIE", "x": [ 24, 47 ], "y": [ 4, 23 ], "density": 0.35 },
+        { "monster": "GROUP_ZOMBIE", "x": [ 24, 47 ], "y": [ 24, 47 ], "density": 0.35 },
+        { "monster": "GROUP_ZOMBIE", "x": [ 24, 47 ], "y": [ 48, 71 ], "density": 0.35 },
+        { "monster": "GROUP_ZOMBIE", "x": [ 24, 47 ], "y": [ 72, 95 ], "density": 0.3 }
+      ]
+    }
+  }
```
Comment (nexusmrsep) [inline data/json/mapgen/town_hall.json:197]: Shouldn't there be a coma here?

Comment (JonathanLochridge) [reply data/json/mapgen/town_hall.json:197]: I tried it both with and without a comma at the end of each block of the map. And it didn't affect whether it passed the linter.
Unless you are referring to another place?

I left it at off and tried changing a few other minor things that didn't fix it.
I realized I forgot to remove the definitions indicating the duplicate roof tiles So I already removed those.

**Comment by JonathanLochridge** [comment]
(I thought I already posted this but I forgot to.)
The following error occurs when I run the 2nd newest experimental with the included changes. Then it gets stuck while trying to open up the create characters screen on the inventory items point. Even though I didn't touch that.
When I hit play now it sometimes crashes instead of freezing. 
The same error occurs for all of the new tile maps for the new town hall.
 
 DEBUG    : Mapgen town_hall_2_0_0_0 is not used by anything!

 FUNCTION : void mapgen_factory::check_consistency()
 FILE     : src/mapgen.cpp
 LINE     : 435
 VERSION  : f1997bf

**Comment by LovamKicsiGazsii** [comment]
Since you didn't change the town hall too drastically, I think duplicating the entire building is a bit overkill.  If you only want to make a variant of one om_terrain, you could take advantage of the weight system and assign two mapgens on the same overmap terrain.  That way, you wouldn't have to edit regional_map_settings.json, multitile_city_buildings.json, and overmap_terrain_public_institutional.json.

Btw, nice addition!

**Comment by JonathanLochridge** [comment]
@LovamKicsiGazsii I am not exactly sure how to do that? My biggest change is the the second story but I had small changes to the first floor. I thought about trying to split up the given map so that I only had to include the changed sections. But, I couldn't even get it to work together. And If I cut it out at the wrong point it would probably mess it up.
I have a couple more ambitious ideas. But, I wanted to make sure I could get a simpler one to work first. 

It still doesn't work. Not sure why it crashes? I can't figure out why? 
It would be nice if someone was willing to help me figure out why. But, I don't expect anyone to. So for now I have been working on other stuff.

**Comment by JonathanLochridge** [comment]
I did some checking and apparently there aren't any crashes on the newest experimental version and it can be debug spawned properly but it doesn't seem to naturally occur at normal spawn rates. Although, maybe I was just unlucky? Since when I ramped up the spawn rate to a higher level it spawned without error. 
I do get an error on new game but it seems to be a different one and doesn't impede spawning. 
I also want to add the new varient to the list of spawn options for the town hall start. however, Since I forked this branch before that change was merged it might be easier to put that in it's own PR?
The debug line below doesn't seem to to be the same coordinates as where the new town hall varient spawned so it may be unrelated. If someone could give me an idea if this error occurs sometimes without my change then I think it should be ready to merge.
 DEBUG    : loading non-main map at (462,610,0) which overlaps with main map (abs_sub = (459,607,0)) is not supported

 FUNCTION : void map::load(const tripoint_abs_sm&, bool, bool)
 FILE     : src/map.cpp
 LINE     : 6710
 VERSION  : 761fb51

**Pull request closed without merging**